product/models.py

```diff
@@ -36,6 +36,8 @@ def save_product(self):
     @staticmethod
     def delete_product(prd_qr):
         try:
-            Product.filter_qr(prd_qr).delete()
-        except Exception:
-            pass
+            prod = Product.objects.get(qr_code=prd_qr)
+            prod.delete()
+        except Product.DoesNotExist as e:
```

Comment:
With the current implementation, I'm not sure I understand the idea behind having this function anymore (that also relates to what Barak mentioned in class on one of the other team's presentations).

What's the benefit of using this static method comparing to the built-in `delete` member method?
It just adds code that, in the best case, does exactly what the built in code does but requires you to maintain additional code.

It's better to use preexisting code (which is someone else's problem), especially if it comes from a library you already import on your codebase, comparing to start maintaining additional code that does the same.

Reply:
Fixed